### PR TITLE
WP-72 Fix UTF-8 encoded characters when passing URLs through to the api.

### PR DESCRIPTION
**Pull request by willh-meetup**
This fixes the problem with the API not recognizing group URLs when calling through the api. We were encode the params, but not the URL itself.

### CODE REVIEW

File: src/apiProxy/api-proxy.test.js [hunk 1]
```diff
@@ -94,6 +95,17 @@ describe('buildRequestArgs', () => {
...
+	});
+
+	const testQueryResults_utf8 = mockQuery(MOCK_RENDERPROPS_UTF8);
+	const apiConfig_utf8 = queryToApiConfig(testQueryResults_utf8);
+	const encoded_url = '/%E3%83%90-%E4%BA%AC';
+
+	it('Properly encodes the URL', () => {
+		const method = 'get';
+		const getArgs = buildRequestArgs({ ...options, method })(apiConfig_utf8);
+		expect(getArgs.url.split('?')[0]).toEqual(encoded_url);
```
Comment (willh-meetup) [inline src/apiProxy/api-proxy.test.js]: not 100% sure how I feel about this test. 

I think it's a good idea to make sure the URL is fully encoded, but maybe there's a better way to do this.

File: src/apiProxy/api-proxy.js [hunk 2]
```diff
@@ -99,7 +99,7 @@ function urlFormatParams(params, doEncode) {
  */
 export const buildRequestArgs = externalRequestOpts => ({ endpoint, params }) => {
 	const externalRequestOptsQuery = { ...externalRequestOpts };
-	externalRequestOptsQuery.url = `/${endpoint}`;
+	externalRequestOptsQuery.url = encodeURI(`/${endpoint}`);
```
Comment (willh-meetup) [inline src/apiProxy/api-proxy.js]: This seemed like the best place to make sure this is always encoded, and I can't think of a reason we wouldn't want to do this before sending along to the API.

File: src/apiProxy/api-proxy.test.js (hunk 1 above)
Comment (mmcgahan) [inline src/apiProxy/api-proxy.test.js]: I think a slightly more general test would parse the URL for the `pathname` and then test that against an [isASCII regex](http://stackoverflow.com/a/14313213/579685)

```js
const { pathname } = require('url').parse(getArgs.url);
expect(/^[\x00-\xFF]*$/.test(pathname)).toBe(true);
```

Comment (willh-meetup) [inline src/apiProxy/api-proxy.test.js]: 👍

**Pull request merged**